KMaina/Fast-Food-Fast-API
# #160784176 Add JWT Functionality to the App and on Certain Routes

#### What does this PR do?
Add JWT functionality for authorization
#### Description of Task to be completed?
Have a token generated and only allow a user to access certain endpoints if the token is valid
#### How should this be manually tested?

- Clone the repo locally.

- cd into Fast-Food-Fast-API folder

- Create a `.env` file and add the following configurations

```
export FLASK_APP = run.py
export FLASK_ENV = development
export SECRET_KEY = 'Anything-you-want'
```

- Create a virtual environment using the command `virtualenv <environment name>`

- Run the app using the command `flask run`
- Navigate to the `POST api/v1/users` endpoint and use the JSON object below
```
{
            "name" = "Ken Maina", 
            "password" =  "Passing", 
            "username" = "Ken",
            "email" =  "ken@me.com",
            "address" =  "Langata, Nairobi",
            "telephone" =  "+722222222",
            "admin" = true
}
```
- Navigate to the `POST api/v1/users/login` endpoint and use the JSON object below
```
{
    "username" : "Ken",
    "password" : "Passing"
}
```

- Grab the token generated in order to use it as a header to ensure that you are authenticated
- Navigate to `GET api/v1/users` endpoint, set the token and you are authorized

#### What are the relevant pivotal tracker stories?
#160784176
